micro-rdk/src/esp32/nvs_storage.rs

```diff
@@ -126,6 +126,40 @@ impl NVSStorage {
     }
 }
 
+impl StorageDiagnostic for NVSStorage {
+    type Error = NVSStorageError;
+    fn log_space_diagnostic(&self) -> Result<(), Self::Error> {
+        let part_name = CString::new("nvs")
+            .map_err(|_| EspError::from_non_zero(NonZeroI32::new(ESP_ERR_INVALID_ARG).unwrap()))?;
```

Comment:
should we be unwrapping here?

Reply:
Yes, I think this is an appropriate use of unwrap, if this panics it means there is something serious wrong with ESP-IDF